backend/tracim_backend/models/context_models.py

```diff
@@ -726,17 +726,49 @@ def frontend_url(self):
         )
         return root_frontend_url + content_frontend_url
 
+    # file specific
+    @property
+    def nb_pages(self):
+        if self.content.depot_file:
+            from tracim_backend.lib.core.content import ContentApi
+            content_api = ContentApi(
+                current_user=self._user,
+                session=self.dbsession,
+                config=self.config
+            )
+            return content_api.get_preview_nb_pages(self.content.revision_id)
+        else:
+            return None
+
+    @property
+    def mimetype(self):
+        return self.content.file_mimetype
+
+    @property
+    def size(self):
+        if self.content.depot_file:
+            from tracim_backend.lib.core.content import ContentApi
+            content_api = ContentApi(
+                current_user=self._user,
+                session=self.dbsession,
+                config=self.config
+            )
+            return content_api.get_file_size(self.content.revision_id)
+        else:
+            return None
```

Comment:
Is it really required than page_nb or size return always int ? 
Allowing None/null value is only a problem in case we explicitly need to use this value in query parameter. None case has a clearer meaning than -1 and obviously 0 for this case where page_nb and size value doesn't make sense.

Reply:
@inkhey the problem I see is : 

```
if not obj.page_nb:
    # do some stuff
```

would be used to detect None cases but 0 is considered as a null case.

Or you would write 

```
if obj.page_nb == None:
    # do some stuff
```

And pep8 will suggest to change this - and a developer will end with doing the refactor which will result in a bug.

@buxx @inkhey let's talk together (the 2 of you) and decide what to do ; then comment this discussion to tell what you decided

Reply:
We choose to used None because those objects are normally only used to serve for marshmallow schema and having a api returning "null" in json is clearly better than '-1' .

An as far i know, the correct way to check None value is "if obj.page_nb is None" but that right that 'if not" statement may be used incorrectly.